packages/next-codemod/bin/upgrade.ts

```diff
@@ -172,12 +172,24 @@ export async function runUpgrade(
 
   fs.writeFileSync(appPackageJsonPath, JSON.stringify(appPackageJson, null, 2))
 
-  const nextDependency = `next@${targetNextVersion}`
-  const dependenciesToInstall = [
-    `react@${targetReactVersion}`,
-    `react-dom@${targetReactVersion}`,
-  ]
+  const dependenciesToInstall = []
   const devDependenciesToInstall = []
+
+  const corePackageNameVersionMapping = {
+    react: `react@${targetReactVersion}`,
+    'react-dom': `react-dom@${targetReactVersion}`,
+    next: `next@${targetNextVersion}`,
+  }
+  for (const packageName of Object.keys(corePackageNameVersionMapping)) {
+    // First check if it's dev dependencies, then add into dev deps;
+    // If it's not, fallback into dependencies.
+    if (appPackageJson.devDependencies?.[packageName]) {
+      devDependenciesToInstall.push(corePackageNameVersionMapping[packageName])
+    } else {
+      dependenciesToInstall.push(corePackageNameVersionMapping[packageName])
+    }
```

Comment:
Let's check for `peerDependencies` as well. I just thought about how this command works for libraries. Unless there's something obvious, we can start thinking about it here.

Reply:
Actually, nvm. `peerDependencies` are a bit trickier. We can expect from library maintainers to get this right.